File: scheduler-core/src/main/java/com/averude/uksatse/scheduler/core/dto/SummationDTO.java
package com.averude.uksatse.scheduler.core.dto;

import com.averude.uksatse.scheduler.core.entity.Employee;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.List;

@Getter
@Setter
public class SummationDTO extends BasicDto<Employee, SummationResult> implements Serializable {
    private LocalDate from;
    private LocalDate to;

    public SummationDTO(Employee parent,
                        LocalDate from,
                        LocalDate to,
                        List<SummationResult> collection) {
        super(parent, collection);
        this.from = from;
        this.to = to;
    }
}

File: scheduler-frontend-server/src/app/model/dto/summation-dto.ts
import { Employee } from "../employee";

export class SummationDto {
  employee: Employee;
  from:     string;
  to:       string;
  results:  SummationResult[];
}

export class SummationResult {
  summationColumnId:  number;
  type:               string;
  value:              number;
}
